src/components/molecules/NavBar/components/NavBarLogin/NavBarLogin.tsx

```diff
@@ -1,28 +1,38 @@
-import React, { FC, useCallback } from "react";
-import { useHistory } from "react-router-dom";
+import { FC, useCallback } from "react";
 
-import { DEFAULT_SPACE_SLUG, DEFAULT_WORLD_SLUG } from "settings";
+import {
+  ATTENDEE_SPACE_ENTRANCE_URL,
+  ATTENDEE_SPACE_INSIDE_URL,
+} from "settings";
 
-import { venueEntranceUrl } from "utils/url";
+import { generateUrl } from "utils/url";
 
 import { useSpaceParams } from "hooks/spaces/useSpaceParams";
 
 import "./NavBarLogin.scss";
 
-export const NavBarLogin: FC = () => {
-  const history = useHistory();
+export interface NavBarLoginProps {
+  hasEntrance: boolean;
+}
+
+export const NavBarLogin: FC<NavBarLoginProps> = ({ hasEntrance }) => {
   const { worldSlug, spaceSlug } = useSpaceParams();
 
-  const navigateToDefault = useCallback(
-    () =>
-      history.push(
-        venueEntranceUrl(
-          worldSlug ?? DEFAULT_WORLD_SLUG,
-          spaceSlug ?? DEFAULT_SPACE_SLUG
-        )
-      ),
-    [history, worldSlug, spaceSlug]
-  );
+  const navigateToDefault = useCallback(() => {
+    const redirectUrl = generateUrl({
+      route: !hasEntrance
+        ? ATTENDEE_SPACE_INSIDE_URL
+        : ATTENDEE_SPACE_ENTRANCE_URL,
+      required: ["worldSlug", "spaceSlug"],
+      params: { worldSlug, spaceSlug, step: "1" },
+    });
+
+    /**
+     * Using history.push() doesn't work correctly for some strange reason and firebase returns 403 (Missing or insufficient premissions).
+     * It's almost like it doesn't detect when the user logs in and the rules consider the `uid` to be `null`
```

Comment:
I think I know what the issues is. The above comment summed up is:

React doesn't refresh the state (from DB, in the Redux cache or elsewhere) before it tries to re-route, but `window.location.href =` does make the page refresh, thus the login state is updated.

In order to continue using `history.push()` we might need to see if the above assumption is correct and if so, how to force the state refresh before the route is changed. Possibly by using `async` or `.then()`.

Reply:
The problem is in the `VenuePage.tsx` it renders login form when user is missing and content when user exists. However the firebase rules doesn't seem to detect the change only on re-render and that's why the page is now reloaded after login.